Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -0,0 +1,785 @@
+-- Returns a fully qualified name thatincludes the package, hash, and name, e.g.
+--
+-- "containers-0.5.6.2@conta_2C3ZI8RgPO2LBMidXKTvIU:Data.Map.Base.fromList".
+qualifiedName
+    :: forall m. (GhcMonad m, MonadIO m, GmOut m, GmLog m)
+    => String -> Int -> Int -> String -> [String] -> m [String]
+qualifiedName targetModuleName lineNr colNr symbol importList = do
+        setContext (map (IIDecl . simpleImportDecl . mkModuleName) (targetModuleName:importList))
+           `gcatch` (\(s  :: SourceError)    -> do gmLog GmDebug "" $ strDoc $ "qualifiedName: setContext failed with a SourceError, trying to continue anyway..." ++ show s
+                                                   setContext $ map (IIDecl . simpleImportDecl . mkModuleName) importList)
+           `gcatch` (\(g  :: GhcApiError)    -> do gmLog GmDebug "" $ strDoc $ "qualifiedName: setContext failed with a GhcApiError, trying to continue anyway..." ++ show g
+                                                   setContext $ map (IIDecl . simpleImportDecl . mkModuleName) importList)
+           `gcatch` (\(se :: SomeException)  -> do gmLog GmDebug "" $ strDoc $ "qualifiedName: setContext failed with a SomeException, trying to continue anyway..." ++ show se
+                                                   setContext $ map (IIDecl . simpleImportDecl . mkModuleName) importList)
+
+        modSummary <- getModSummary $ mkModuleName targetModuleName :: m ModSummary
+        p <- parseModule modSummary                                 :: m ParsedModule
+        t <- typecheckModule p                                      :: m TypecheckedModule
+
+        let TypecheckedModule{tm_typechecked_source = tcs} = t
+            bs = listifySpans tcs (lineNr, colNr) :: [LHsBind Id]
+            es = listifySpans tcs (lineNr, colNr) :: [LHsExpr Id]
+            ps = listifySpans tcs (lineNr, colNr) :: [LPat Id]
+
+        let bs' = map (showSDocForUser tdflags reallyAlwaysQualify . ppr) bs
+            es' = map (showSDocForUser tdflags reallyAlwaysQualify . ppr) es
+            ps' = map (showSDocForUser tdflags reallyAlwaysQualify . ppr) ps
+
+        return $ filter (postfixMatch symbol) $ concatMap words $ bs' ++ es' ++ ps'
+
+
+
+ghcPkgFindModule
+    :: forall m. (GhcMonad m, MonadIO m, GmOut m, GmLog m)
+    => String
+    -> m (Maybe String)
+ghcPkgFindModule mod
+    = shortcut [ stackGhcPkgFindModule mod
+               , hcPkgFindModule       mod
+               , _ghcPkgFindModule     mod
+               ]
+  where
+    shortcut :: [m (Maybe a)] -> m (Maybe a)
+    shortcut []     = return Nothing
+    shortcut (a:as) = do
+        a' <- a
+
+        case a' of
+            a''@(Just _)    -> return a''
+            Nothing         -> shortcut as
+
+    executeFallibly' :: String -> [String] -> IO (Maybe (String, String))
+    executeFallibly' cmd args = do
+        x <- executeFallibly (piped (proc cmd args)) ((,) <$> foldOut intoLazyBytes <*> foldErr intoLazyBytes)
```

Comment:
S.P.Streaming seems quite pointless for, you could just use `readProcessWithExitCode`. Which makes the whole roundtrip through ByteString pointless too. Given you later just print stderr please just use `gmReadProcess` which returns a function that handles translating stderr for `--line-prefix` stuff.